### PR TITLE
Creation of tasks for using event-shape engineering with jets

### PR DESCRIPTION
**Pull request by caitiebeattie**
Addition of AliJEQnVectorHandler.cxx, AliAnalysisTaskJetQnVectors.cxx, and AddTaskJEQnVectors.C to calculate Qn vectors in jet events. Modification of AliAnalysisTaskJetExtractor.cxx to access q2 vectors.

### CODE REVIEW

File: PWGJE/EMCALJetTasks/CMakeLists.txt [hunk 1]
```diff
@@ -295,6 +296,7 @@ if(FASTJET_FOUND)
         UserTasks/AliAnalysisTaskHJetSpectra.cxx
         UserTasks/AliAnalysisTaskIDFragmentationFunction.cxx
         UserTasks/AliHelperClassFastSimulation.cxx
+	UserTasks/AliJEQnVectorHandler.cxx
```
Comment (raymondEhlers) [inline PWGJE/EMCALJetTasks/CMakeLists.txt:299]: Trivial, but please match the indentation

File: PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskJetQnVectors.h [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+#include <TList.h>
+#include <TString.h>
+#include <TMath.h>
+
+#include "AliAnalysisTaskSE.h"
+#include "AliVEvent.h"
+#include "AliAODEvent.h"
+#include "AliJEQnVectorHandler.h"
+
+using namespace std;
```
Comment (raymondEhlers) [inline PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskJetQnVectors.h:33]: Please move this to your .cxx. `using namespace std` is generally to be avoided in a header

File: PWGJE/EMCALJetTasks/UserTasks/AliJEQnVectorHandler.h [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+#include <TH1D.h>
+#include <TString.h>
+
+#include "AliAODEvent.h"
+#include "AliAODTrack.h"
+#include "AliAODVZERO.h"
+#include "AliAnalysisTaskFlowVectorCorrections.h"
+#include "AliQnCorrectionsManager.h"
+
+using namespace std;
```
Comment (raymondEhlers) [inline PWGJE/EMCALJetTasks/UserTasks/AliJEQnVectorHandler.h:35]: Same here

**Review by raymondEhlers — COMMENTED** [review]
Hi Caitie, I only had time for a very quick look, but looks good to me. I have just a few small comments. I didn't really look at the QnVector and Handler classes because as far as I can tell, you're basically copying them from somewhere else and adapting them, so I assume that they're fine

**Comment by caitiebeattie** [comment]
Thanks for the comments @raymondEhlers , I have made the suggested changes.

**Comment by raymondEhlers** [comment]
Thanks Caitie - please squash all of your commits into one, and then I'll approve

**Comment by caitiebeattie** [comment]
@raymondEhlers I've squashed them. Thanks!

**Comment by raymondEhlers** [comment]
+1

**Pull request merged**